### PR TITLE
Pubsub inrepoconfig presubmit

### PR DESCRIPTION
**Pull request by listx**
The fake (nonsense) cookie file we mount is only there to trigger the
code branch that detects whether to create a git client factory for
GitHub or for Gerrit. The --cookie-file flag makes sub's ProwYAML cache
initialization logic assume that all inrepoconfigs are cloned from
Gerrit.

For our purposes the contents of the cookie file
doesn't matter because we don't even hit the fakegerritserver URL for
cloning the repo because we use a direct CloneURI field.

Note that this tightly couples this deployment of sub to Gerrit (see the
TODO comment added in sub.yaml).

In the future if we want to add a test case without using the CloneURI
field, we would have to refactor how we initialize the inrepoconfig
cache (in particular how we set up the git client factories). Currently
we either always clone from GitHub or Gerrit (by prefixing "https://"
--- see tryGetCloneURIAndHost() in
prow/pubsub/subscriber/subscriber.go). Obviously neither choice will
work because we need to clone instead from
"http://fakegitserver.default/" ... for the integration tests to work.

/assign @mpherman2 
/cc @chaodaiG @cjwagner

### CODE REVIEW

**Comment by mpherman2** [comment]
/hold
in case you want others to look at it. Feel free to unhold

File: prow/test/integration/test/sub_test.go
Comment (chaodaiG) [inline prow/test/integration/test/sub_test.go:192]: can you replace this with `Org: "http://fakegitserver.default/repo/org1",`? I think sub should be smart enough to clone `http://fakegitserver.default/repo/org1/repos`

File: prow/pubsub/subscriber/subscriber.go
Comment (listx) [reply prow/pubsub/subscriber/subscriber.go:484]: Moved out to https://github.com/kubernetes/test-infra/pull/26574

File: prow/test/integration/test/sub_test.go
Comment (chaodaiG) [reply prow/test/integration/test/sub_test.go:192]: Once #26574 is merged, this should pass. Would recommend add another test case as I mentioned above, iirc sub knows to concatenate `<ORG>/<REPO>` for cloning, so that even if there is no `CloneURI` it would also work.

Comment (listx) [reply prow/test/integration/test/sub_test.go:192]: I think you're right. Let me update this PR after I confirm that the integration tests still pass locally.

Comment (listx) [reply prow/test/integration/test/sub_test.go]: I think there's a little more subtlety involved because of the special meaning of `https://` WRT Gerrit-style org names. We've used `http://` so far in these tests, which is not the same and is not what we expect in production anyway. So I'm going to try to make the tests use https style Org names and see how that goes.

Comment (listx) [reply prow/test/integration/test/sub_test.go]: I've created https://github.com/kubernetes/test-infra/pull/26587 to clean this part up a little bit, so that we can use "https://..." style refs everywhere in our integration tests. Once we merge that I can update these tests.

**Comment by listx** [comment]
FTR I had to update the timeout of the test cases added here from 30s to 60s to make these tests pass. I think running other tests in parallel puts additional stress on the system that is not observed when running these inrepoconfig tests in isolation (I could repro the failures observed earlier by deleting my local test cluster and rerunning everything from scratch with a 30s timeout).

File: prow/test/integration/test/sub_test.go
Comment (listx) [reply prow/test/integration/test/sub_test.go]: Done(?). PTAL

File: prow/test/integration/test/sub_test.go
Comment (listx) [reply prow/test/integration/test/sub_test.go:334]: > read this comment three times and then realized what it means, might be better phrase it in this way:

Sure I can do that.

> what you do you think about having a test with an inrepoconfig job that doesn't have cloneUri specified?

That's what the ProwJobDecorated const does.

**Review by chaodaiG — APPROVED** [review]

**Pull request merged**
